sdk/schemaregistry/azure-schemaregistry/azure/schemaregistry/_schema_registry_client.py

```diff
@@ -0,0 +1,180 @@
+    :param credential: To authenticate to manage the entities of the SchemaRegistry namespace.
+    :type credential: TokenCredential
+
+    .. admonition:: Example:
+
+        .. literalinclude:: ../samples/sync_samples/sample_code_schemaregistry.py
+            :start-after: [START create_sr_client_sync]
+            :end-before: [END create_sr_client_sync]
+            :language: python
+            :dedent: 4
+            :caption: Create a new instance of the SchemaRegistryClient.
+
+    """
+    def __init__(
+        self,
+        endpoint,
+        credential,
+        **kwargs
+    ):
+        # type: (str, TokenCredential, Any) -> None
+        self._generated_client = AzureSchemaRegistry(credential=credential, endpoint=endpoint, **kwargs)
+
+    def __enter__(self):
+        # type: () -> SchemaRegistryClient
+        self._generated_client.__enter__()
+        return self
+
+    def __exit__(self, *exc_details):
+        # type: (Any) -> None
+        self._generated_client.__exit__(*exc_details)
+
+    def close(self):
+        # type: () -> None
+        """ This method is to close the sockets opened by the client.
+        It need not be used when using with a context manager.
+        """
+        self._generated_client.close()
+
+    def register_schema(self, schema_group, schema_name, serialization_type, schema_content, **kwargs):
+        # type: (str, str, Union[str, SerializationType], str, Any) -> SchemaProperties
+        """
+        Register new schema. If schema of specified name does not exist in specified group,
+        schema is created at version 1. If schema of specified name exists already in specified group,
+        schema is created at latest version + 1.
+
+        :param str schema_group: Schema group under which schema should be registered.
+        :param str schema_name: Name of schema being registered.
+        :param serialization_type: Serialization type for the schema being registered.
+        :type serialization_type: Union[str, SerializationType]
+        :param str schema_content: String representation of the schema being registered.
+        :rtype: SchemaProperties
+
+        .. admonition:: Example:
+
+            .. literalinclude:: ../samples/sync_samples/sample_code_schemaregistry.py
+                :start-after: [START register_schema_sync]
+                :end-before: [END register_schema_sync]
+                :language: python
+                :dedent: 4
+                :caption: Register a new schema.
+
+        """
+        try:
+            serialization_type = serialization_type.value
+        except AttributeError:
+            pass
+
+        return self._generated_client.schema.register(
+            group_name=schema_group,
+            schema_name=schema_name,
+            schema_content=schema_content,
+            #serialization_type=serialization_type,  # TODO: current swagger doesn't support the parameter
```

Comment:
Why do we expose a parameter that is not used? It's just confusing for people

Reply:
Hmm, the current swagger and service implementation does take this argument. It's currently passed as x-schema-type header. It is going to be renamed to "serialization type" from "schema type". I'm surprised this works. In my tests, it did not work unless x-schema-type=avro was sent. Is something in the generated client defaulting that now, did that change, or am I misremembering.

Anyway, in JS, I send the serializationType arg as the x-schema-type.

cc @arerlend

Reply:
thanks @nguerrera , I almost forget this parameter, will test this ASAP.

Reply:
using the latest python autorest to will generate the expected code.
code updated.